src/main/java/org/folio/inventory/dataimport/consumers/QuickMarcKafkaHandler.java

```diff
@@ -93,18 +88,14 @@ public QuickMarcKafkaHandler(Vertx vertx, Storage storage, int maxDistributionNu
     var params = new OkapiConnectionParams(kafkaHeadersToMap(record.headers()), vertx);
     var context = constructContext(params.getTenantId(), params.getToken(), params.getOkapiUrl());
     Event event = Json.decodeValue(record.value(), Event.class);
-    if (!kafkaInternalCache.containsByKey(event.getId())) {
-      LOGGER.info(format("Quick marc event payload has been received with event type: %s", event.getEventType()));
-      kafkaInternalCache.putToCache(event.getId());
-      return getEventPayload(event)
-        .compose(eventPayload -> processPayload(eventPayload, context)
-          .compose(recordType -> sendEvent(eventPayload, getReplyEventType(recordType), params))
-          .recover(throwable -> sendErrorEvent(params, eventPayload, throwable))
-          .map(ar -> record.key()),
-          th -> Future.failedFuture(th.getMessage())
-        );
-    }
-    return Future.succeededFuture();
+    LOGGER.info("Quick marc event payload has been received with event type: {}", event.getEventType());
+    return getEventPayload(event)
+      .compose(eventPayload -> processPayload(eventPayload, context)
+        .compose(recordType -> sendEvent(eventPayload, getReplyEventType(recordType), params))
+        .recover(throwable -> sendErrorEvent(params, eventPayload, throwable))
+        .map(ar -> record.key()),
+        th -> Future.failedFuture(th.getMessage())
```

Comment:
Just to double check, is string with exception message for this failed future logged in some place?

Reply:
Good point! Thanks